### DISCUSSION TITLE
Negative Numbers/Units.

### DISCUSSION THREAD

**Discussion by astsbins** [iSoron/uhabits#1385] [Feature Requests]
I think it would be really nice to have the ability to be able to have the units for measurable to goals to be in units. Say for example may goal is to read 10 pages of a book. If i read 15 pages, great! I can put in 5. However, if say I only manage to read 5, it would be good to be able -5 instead of just 0.

**Comment by hunterwilhelm** [comment]
I too would also like negative numbers to track bad habits.

**Comment by nitobuendia** [comment]
Another use case: Tracking monetary quantities - e.g. tracking savings. You may save $X one day/week/month, but you may spend $Y from your savings.

**Comment by nathou** [comment]
I would also love that, to count the number of minutes I go to bed before or after a certain time. The target would be 0, going before would be a positive number and after a negative number. I currently have two habits for that (one negative and one positive) but it would be great to have only one.

**Reply by Cancoillotte** [reply]
I'm glad to see that I'm far from being the only one to ask this question. I've been looking for a long time at how to do it, but you're obliged to have a 0, whatever time you choose, and so you may have to put a negative number, because it can always happen that you have to go to bed before the cut-off time. If you can't put a negative number, it distorts the average. At present, I indicate in the comments the number of minutes before the cut-off time when it's “negative”, but it's really just a matter of tinkering.

@iSoron , would it be possible to implement this negative numbers feature? Thanks in advance for your attention!

**Reply by herrbrixter** [reply]
I also like this idea. This is related to #953. It should be possible to input negative time duration. For example, "-0:20" means you completed an activity 20 minutes faster than normal.

**Comment by isgin01** [comment]
I would love this app to have this feature too. I track days' quality and it would be more convenient for me to be able to put -1, 0, 1 values instead of 1, 2, 3.

**Comment by rohitdwivedula** [comment]
+1, I think this would be useful for habits like "calorific defecit", for example.

**Reply by 4nn1nw0nd3rl4nd-hash** [reply]
That's exactly what I'm looking for. I love the app and use it to track all my habits but I miss a way to track if I stayed in my caloric deficit. At the moment I use a goal with minimum zero but it doesn't track days well where I ate less calories. If anyone got an idea how to track this I would be really gratefull. Negative numbers would of course do the trick but I don't know if this feature will be implemented soon.

**Comment by qianxiaoning** [comment]
+1. Yes, if quantifiable habits supported negative numbers, it would be much more convenient. For example, when tracking remaining available time >0 is positive and ≤0 is negative. A single habit could record both positive and negative states at the same time. Currently, when my remaining available time is negative, I can only write the negative value in the notes. The existing quantifiable value cannot reflect the specific number, so I have to just write 0 every day.
